components/homme/cmake/SetCompilerFlags.cmake

```diff
@@ -67,11 +67,11 @@ IF (${HOMME_USE_CXX})
   SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
 
   INCLUDE(CheckCXXCompilerFlag)
-  CHECK_CXX_COMPILER_FLAG("-std=c++14" CXX14_SUPPORTED)
-  IF (${CXX14_SUPPORTED})
-    SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
+  CHECK_CXX_COMPILER_FLAG("-std=c++17" CXX17_SUPPORTED)
+  IF (${CXX17_SUPPORTED})
+    SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
```

Comment:
Are we sure C++17 is actually needed? I had the impression that it was thought Crusher requires it but then it turns out it does not.

Reply:
Now I do not remember why it is here. Do you prefer 14?

Reply:
For cleanliness we should not mandate more than is technically required. I believe for a while weaver could not handle C++17, but that may have changed now.